packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart

```diff
@@ -0,0 +1,216 @@
+  /// Creates a project which uses a plugin, which is not supported on Android.
+  /// This means it has no entry in pubspec.yaml for:
+  /// flutter -> plugin -> platforms -> android
+  ///
+  /// [createAndroidPluginFolder] indicates that the plugin can additionally
+  /// have a functioning `android` folder.
+  Future<ProcessResult> testUnsupportedPlugin({
+    required Project project,
+    required bool createAndroidPluginFolder,
+  }) async {
+    final String flutterBin = fileSystem.path.join(
+      getFlutterRoot(),
+      'bin',
+      'flutter',
+    );
+
+    // Create dummy plugin that supports iOS and optionally Android.
+    processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'create',
+      '--template=plugin',
+      '--platforms=ios${createAndroidPluginFolder ? ',android' : ''}',
+      'test_plugin',
+    ], workingDirectory: tempDir.path);
+
+    final Directory pluginAppDir = tempDir.childDirectory('test_plugin');
+
+    final File pubspecFile = pluginAppDir.childFile('pubspec.yaml');
+    String pubspecYamlSrc =
+        pubspecFile.readAsStringSync().replaceAll('\r\n', '\n');
+    if (createAndroidPluginFolder) {
+      // Override pubspec to drop support for the Android implementation.
+      pubspecYamlSrc = pubspecYamlSrc
+          .replaceFirst(
+        RegExp(r'name:.*\n'),
+        'name: test_plugin\n',
+      )
+          .replaceFirst('''
+      android:
+        package: com.example.test_plugin
+        pluginClass: TestPlugin
+''', '''
+#      android:
+#        package: com.example.test_plugin
+#        pluginClass: TestPlugin
+''');
+
+      pubspecFile.writeAsStringSync(pubspecYamlSrc);
+
+      // Check the android directory and the build.gradle file within.
+      final File pluginGradleFile =
+          pluginAppDir.childDirectory('android').childFile('build.gradle');
+      expect(pluginGradleFile, exists);
+    } else {
+      expect(pubspecYamlSrc, isNot(contains('android:')));
+    }
+
+    // Create a project which includes the plugin to test against
+    final Directory pluginExampleAppDir =
+        pluginAppDir.childDirectory('example');
+
+    await project.setUpIn(pluginExampleAppDir);
+
+    // Run flutter build apk to build plugin example project.
+    return processManager.runSync(<String>[
+      flutterBin,
+      ...getLocalEngineArguments(),
+      'build',
+      'apk',
+      '--debug',
+    ], workingDirectory: pluginExampleAppDir.path);
+  }
+
+  test('skip plugin if it does not support the Android platform', () async {
+    final Project project = PluginWithPathAndroidProject();
+    final ProcessResult buildApkResult = await testUnsupportedPlugin(
+        project: project, createAndroidPluginFolder: false);
+    expect(buildApkResult.stderr.toString(),
+        isNot(contains('Please fix your settings.gradle.')));
+    expect(buildApkResult.exitCode, equals(0),
```

Comment:
nit please use https://github.com/flutter/flutter/blob/37510051329dcfc962db8b6de80c3ef5e7b639fe/packages/flutter_tools/test/integration.shard/test_utils.dart#L136, it will handle constructing a failure reason with the actual exit code, and actual stdout/stderr

Reply:
Thanks didn't know about that. I have to figure out the regex for NOT containing something.

Reply:
I would leave the separate stderr isNot(contains()) expectation. Even if you COULD write that in regex, no one else would be able to read it :). we still get a better failure message on testing the exit code

Reply:
Addressed in 6f79b628f6cb63bd0109f6b09ac10d1d5b2f8f05